src/app/src/main/kotlin/org/imperial/mrc/hint/clients/ADRFuelClient.kt

```diff
@@ -37,9 +41,39 @@ interface ADRClient
 }
 
 class ADRFuelClient(appProperties: AppProperties,
-                    private val apiKey: String)
+                    private val apiKey: String,
+                    private val logger: GenericLogger = GenericLoggerImpl())
     : FuelClient(appProperties.adrUrl + "api/3/action"), ADRClient
 {
+    override fun get(url: String): ResponseEntity<String>
+    {
+        val start = Instant.now()
+        val response = super.get(url)
+        val end = Instant.now()
+        val timeElapsed = Duration.between(start, end).toMillis()
+        logger.info("ADR request time elapsed: $timeElapsed")
```

Comment:
It feels like the time should go into the structured JSON logging as a property valus, rather than just included in the log string like this. That would make it easier to do analytics queries against ADR durations e.g. show me the average ADR fetch duration.